### ISSUE TITLE
Policy update in D4PG

### ISSUE THREAD

**Issue by rasoolfa** [deepmind/acme#56]
Hi,

In D4PG implementation, actor uses s_{t+1} (i.e. next state) to [update](https://github.com/deepmind/acme/blob/799f76fffe8dde4e6bbf3c6cb1c0260cf7ff00aa/acme/agents/tf/d4pg/learning.py#L207) the policy, shouldn't be current state s_t used to update the policy?
```
      dpg_a_t = self._policy_network(o_t)
      dpg_z_t = self._critic_network(o_t, dpg_a_t)
      dpg_q_t = dpg_z_t.mean()
```
instead
```
      dpg_a_t = self._policy_network(o_tm1)
      dpg_z_t = self._critic_network(o_tm1, dpg_a_t)
      dpg_q_t = dpg_z_t.mean()
```

Thanks.

**Comment by bshahr** [comment]
Hi rasoolfa,

Thanks for your question! When optimizing the policy, we need state-action pairs to compute an empirical policy loss that is then minimized to update the policy parameters. We can in fact use either (or both!) of the options you outline above. 

However, one subtlety to be aware of is that currently the policy loss is computed in a way that it does not backpropagate through to the `observation_network` weights (see the `tf.stop_gradient()` on `o_t`). This was a design choice to avoid gradients from the critic loss and policy loss competing when altering the observation network weight (think ResNet). Depending on whether you choose one, the other, or both---and depending on whether you wish to perpetuate this design choice---you will need to carefully place/remove stop-gradients.

Hope that makes sense!

Bobak

**Comment by rasoolfa** [comment]
Second option was referring to the original DDPG's policy update. I thought there is an intuition behind your choice to do next_state instead of current state. But your comments makes sense to me.
Thanks again for your response.
